lib/puppet-strings/yard/parsers/puppet/statement.rb

```diff
@@ -134,13 +134,15 @@ def initialize(object, file)
   # Implements the Puppet function statement.
   class FunctionStatement < ParameterizedStatement
     attr_reader :name
+    attr_reader :type
 
     # Initializes the Puppet function statement.
     # @param [Puppet::Pops::Model::FunctionDefinition] object The model object for the function statement.
     # @param [String] file The file containing the statement.
     def initialize(object, file)
       super(object, file)
       @name = object.name
+      @type = object.return_type.value.capitalize if object.return_type
```

Comment:
Note that `object_type` was introduced quite recently and this will fail if using puppet < 4.7 or so (don't remember exactly when return_type was added). Code should protect against that if Strings can use a puppet older than when it was introduced.

The return type can be a static expression, it is not enough to assume it is a string like value. Needs to do whatever is done for parameter types (have not looked, I am assuming it does it correctly to handle things like Integer[1,10] and other parameterized types).